x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/components/header_actions/search_engines_popover.tsx

```diff
@@ -15,19 +15,54 @@ import {
   EuiContextMenuPanel,
   EuiContextMenuItem,
   EuiText,
+  EuiToolTip,
 } from '@elastic/eui';
 
 import { i18n } from '@kbn/i18n';
 
 import { APP_SEARCH_PLUGIN } from '../../../../../../../common/constants';
+import { ElasticsearchIndexWithIngestion } from '../../../../../../../common/types/indices';
 import { ENGINE_CREATION_PATH } from '../../../../../app_search/routes';
+import { ESINDEX_QUERY_PARAMETER } from '../../../../../shared/constants';
 import { KibanaLogic } from '../../../../../shared/kibana';
+import { addQueryParameter } from '../../../../../shared/query_params';
+import { isHiddenIndex } from '../../../../utils/indices';
 
 import { SearchEnginesPopoverLogic } from './search_engines_popover_logic';
 
-export const SearchEnginesPopover: React.FC = () => {
+export interface SearchEnginesPopoverProps {
+  indexData?: ElasticsearchIndexWithIngestion;
```

Comment:
question: What do you think of having the indexName and isHidden as arguments, rather than popping the full index in there?

Reply:
I originally had `indexName` then just updating it to passing the entire data when I needed to determine if it was hidden. I don't mind having both `indexName` and `isHidden` but wanted to avoid having a list of props here when I thought we would also need to check for the workplace search document indices.